Resources not working

Hi,

I have added a resources section to the configuration to add fanart and subtitles:
```
  <import>
    [...]
    <resources>
      <subtitle>
        <add-file name="%filename%.srt"/>
        <add-file name="%filename%.en.srt"/>
        <add-file name="%filename%.nl.srt"/>
      </subtitle>
      <fanart>
        <add-file name="%filename%-thumb.jpg"/>
        <add-file name="poster.jpg"/>
      </fanart>
      <container>
        <add-file name="poster.jpg"/>
      </container>
      <resource>
        <add-file name="%filename%.srt"/>
        <add-file name="%filename%.en.srt"/>
        <add-file name="%filename%.nl.srt"/>
        <add-file name="%filename%-thumb.jpg"/>
        <add-file name="poster.jpg"/>
      </resource>
    </resources>
  </import>
```

However, when testing with VLC, subtitles don't seem to be loaded.
Fanart works for the movie items themselves, but not for containers.

Loggin:

Fanart found:
```
Jan 28 09:32:58 acer-laptop gerbera[101111]: 2021-01-28 09:32:58  debug: [fanart_handler.cc:56] getFanArtPath(): Folder name: /.../film/Aladdin (2019)
Jan 28 09:32:58 acer-laptop gerbera[101111]: 2021-01-28 09:32:58  debug: [fanart_handler.cc:63] getFanArtPath(): folder.jpg: missing
Jan 28 09:32:58 acer-laptop gerbera[101111]: 2021-01-28 09:32:58  debug: [fanart_handler.cc:63] getFanArtPath(): poster.jpg: found
Jan 28 09:32:58 acer-laptop gerbera[101111]: 2021-01-28 09:32:58  debug: [fanart_handler.cc:87] serveContent(): FanArt: Opening name: /.../film/Aladdin (2019)/poster.jpg
```

Subtitles not found:
```
Jan 28 09:33:03 acer-laptop gerbera[101111]: 2021-01-28 09:33:03  debug: [server.cc:485] createRequestHandler(): Filename: /content/media/object_id/10187/res_id/0/ext/file.srt
Jan 28 09:33:03 acer-laptop gerbera[101111]: 2021-01-28 09:33:03  debug: [file_request_handler.cc:69] getInfo(): start
Jan 28 09:33:03 acer-laptop gerbera[101111]: 2021-01-28 09:33:03  debug: [upnp_clients.cc:218] getInfo(): client info: 192.168.1.191:62534 'VLC/3.0.12 LibVLC/3.0.12' -> 'Unknown' as None
Jan 28 09:33:03 acer-laptop gerbera[101111]: 2021-01-28 09:33:03  debug: [file_request_handler.cc:85] getInfo(): full url (filename): /content/media/object_id/10187/res_id/0/ext/file.srt, parameters: object_id/10187/res_id/0/ext/file.srt
Jan 28 09:33:03 acer-laptop gerbera[101111]: 2021-01-28 09:33:03 warning: [server.cc:533] operator()(): Subtitle file /.../film/Aladdin (2019)/Aladdin (2019) - Bluray-2160p HDR x265 AC3.srt is not available.
```

Folder content:
```
/.../film/Aladdin\ \(2019\)/
total 15G
-rwxrwxrwx 1 root root 133K Jun 27  2020 'Aladdin (2019) - Bluray-2160p HDR x265 AC3.en.srt'
-rwxrwxrwx 2 root root  15G Jun 27  2020 'Aladdin (2019) - Bluray-2160p HDR x265 AC3.mkv'
-rwxrwxrwx 1 root root 7.0K Jan 24 10:40 'Aladdin (2019) - Bluray-2160p HDR x265 AC3.nfo'
-rwxrwxrwx 1 root root 1.7M Jun 26  2020  fanart.jpg
-rwxrwxrwx 1 root root 1.2M Jun 26  2020  poster.jpg
```

I am running Gerbera 1.6.0 from the Ubuntu PPA.

I tried building from source, but cmake is not cooperating.
I built libupnp 1.14.0 from source, but still:
```
> cmake .. -DWITH_MAGIC=1 -DWITH_MYSQL=1 -DWITH_CURL=1 -DWITH_JS=1
[...]
-- Found pupnp: /usr/local/lib/libupnp.so (found suitable version "1.14.0", minimum required is "1.12.1")
CMake Error at cmake/Findpupnp.cmake:50 (set_target_properties):
  set_target_properties called with incorrect number of arguments.
Call Stack (most recent call first):
  CMakeLists.txt:337 (find_package)
```

Also, I did **not** specify `-DWITH_TAGLIB=1 -DWITH_AVCODEC=1 -DWITH_FFMPEGTHUMBNAILER=1 -DWITH_EXIF=1 -DWITH_LASTFM=1`, but cmake still required me to install libmatroska, libtag, libebml and others

--- a/src/metadata/metacontent_handler.cc
+++ b/src/metadata/metacontent_handler.cc
@@ -146,7 +146,7 @@ std::unique_ptr<IOHandler> FanArtHandler::serveContent(std::shared_ptr<CdsObject
     fs::path path = item->getResource(resNum)->getAttribute(R_RESOURCE_FILE);
     if (path.empty()) {
         auto tweak = config->getDirectoryTweakOption(CFG_IMPORT_DIRECTORIES_LIST)->get(item->getLocation());
-        path = getContentPath(tweak == nullptr && !tweak->hasFanArtFile() ? names : std::vector<std::string> { tweak->getFanArtFile() }, item, tweak != nullptr && tweak->hasCaseSensitive() ? tweak->getCaseSensitive() : caseSensitive);
+        path = getContentPath(tweak == nullptr || !tweak->hasFanArtFile() ? names : std::vector<std::string> { tweak->getFanArtFile() }, item, tweak != nullptr && tweak->hasCaseSensitive() ? tweak->getCaseSensitive() : caseSensitive);
     }
     log_debug("FanArt: Opening name: {}", path.c_str());
     struct stat statbuf;

